### PR TITLE
number

### PR DESCRIPTION
**Pull request by jasoncolburne**
## Rationale

Number needs to be completed so that we can create KERI examples.

## Changes

- implements `Number`

## Testing

```
make fix clean preflight
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,9 +6,9 @@ fix:
 	cargo fmt
 
 preflight:
+	cargo audit
```
Comment (jasoncolburne) [inline Makefile:9]: This will fail faster (if it ever does)

File: src/core/indexer/mod.rs [hunk 2]
```diff
@@ -12,7 +12,7 @@ use crate::core::util;
 /// Indexer is fully qualified cryptographic material primitive base class for
 /// indexed primitives. Indexed codes are a mix of indexed and variable length
 /// because code table has two char codes for compact variable length.
-pub(crate) trait Indexer: Default {
+pub trait Indexer: Default {
```
Comment (jasoncolburne) [inline src/core/indexer/mod.rs:15]: I started incorporating Artem's changes into this code, I'll fix the conflicts after he merges his.

File: src/core/number.rs [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+        0xffffffffffffffffffffffffffffffff,
+        "ffffffffffffffffffffffffffffffff",
+        number::Codex::Huge,
+        b"\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff",
+        "0AD_____________________",
+        b"0AD_____________________",
+        b"\xd0\x00\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff",
+        true
+    )]
+    fn happy(
```
Comment (jasoncolburne) [inline src/core/number.rs:287]: these test vectors were all taken from KERIpy

File: src/core/number.rs [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+        "ffffffffffffffffffffffffffffffff",
+        number::Codex::Huge,
+        b"\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff",
+        b"\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff",
+        "0AD_____________________",
+        b"0AD_____________________",
+        b"\xd0\x00\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff",
+        true
+    )]
+    fn truncates(
```
Comment (jasoncolburne) [inline src/core/number.rs:361]: These tests are also from KERIpy

File: src/lib.rs [hunk 5]
```diff
@@ -7,17 +7,23 @@ mod core;
...
+        prefixer::Prefixer,
+        sadder::Sadder,
+        saider::Saider,
+        seqner::Seqner,
+        siger::Siger,
+        signer::Signer,
+        util,
+        verfer::Verfer,
+    },
+    data::Value,
```
Comment (jasoncolburne) [inline src/lib.rs:27]: I'm not sure we need this

File: src/lib.rs [hunk 6]
```diff
@@ -7,17 +7,23 @@ mod core;
...
+        indexer::{tables as indexer, Indexer},
+        matter::{tables as matter, Matter},
+        number::{tables as number, Number},
+        prefixer::Prefixer,
+        sadder::Sadder,
+        saider::Saider,
+        seqner::Seqner,
+        siger::Siger,
+        signer::Signer,
+        util,
```
Comment (jasoncolburne) [inline src/lib.rs:24]: Also not sure we need this.

**Comment by jasoncolburne** [comment]
This isn't ready to merge until after Artem's but I figured it was available so moved to ready.

File: src/lib.rs (hunk 5 above)
Comment (jasoncolburne) [reply src/lib.rs:27]: I think we do, after more work. I needed it for Tholder, Serder.

File: src/lib.rs (hunk 6 above)
Comment (jasoncolburne) [reply src/lib.rs:24]: Could be useful in parside I suppose.

Comment (jasoncolburne) [reply src/lib.rs]: After talking to @m00sey decided to remove this, to minimize the surface we expose

**Review by m00sey — APPROVED** [review]

**Pull request merged**